### PR TITLE
Task/get user info

### PR DESCRIPTION
**Pull request by geekiam23**
## Why?
- Add getUserInfo to retrieve user info by events 
- [NC2-670 -- Add SDK Function for Get User Details from Events API.](https://ndustrialio.atlassian.net/browse/NC2-670)

## What changed?
- Add `getUserInfo` to events
- Update docs to reflect changes

**EventUser Fixture Outcome:**
```
{
  "id": "auth0|16973",
  "createdAt": "2019-03-24T14:21:29.907Z",
  "email": "Ryder.Stark99@hotmail.com",
  "firstName": "Ava",
  "isMachineUser": true,
  "lastName": "Anderson",
  "userMobileNumbers": [],
  "updatedAt": "2020-01-27T22:49:05.793Z",
  "IOSDevice": [
    {
      "userId": "auth0|16973",
      "id": 16,
      "createdAt": "2019-09-03T03:31:13.096Z",
      "deviceToken": "52f4e626-fbb2-4d8d-ba42-5de9369facf4",
      "isActive": true,
      "snsEndpointArn": "e1c477b9-ee72-453c-be85-7f08a1f27550",
      "updatedAt": "2020-01-28T10:02:16.040Z"
    },
    {
      "userId": "auth0|16973",
      "id": 17,
      "createdAt": "2019-05-03T20:44:55.689Z",
      "deviceToken": "80038a00-db7b-4f0b-8167-dbb0b7a9f92a",
      "isActive": true,
      "snsEndpointArn": "677f757a-2e59-40c1-a15c-3bdafdb78dae",
      "updatedAt": "2020-01-28T01:51:52.859Z"
    }
  ],
  "userEventSubscription": [
    {
      "userId": "auth0|16973",
      "createdAt": "2019-06-17T07:01:46.476Z",
      "endpointArn": null,
      "eventId": "dd1ef9eb-5edc-41ff-a7b5-ec09909d4ea5",
      "id": "bf98555c-3757-4ddf-920f-603dd3d5974b",
      "mediumType": "sms",
      "updatedAt": "2020-01-27T20:02:22.390Z"
    },
    {
      "userId": "auth0|16973",
      "createdAt": "2020-01-28T02:17:16.511Z",
      "endpointArn": null,
      "eventId": "ed9f43b4-a242-482d-b4ab-1d0817fbf4fd",
      "id": "38b27bbb-a570-4c7d-9c20-feade7da376f",
      "mediumType": "sms",
      "updatedAt": "2020-01-28T06:42:28.517Z"
    },
    {
      "userId": "auth0|16973",
      "createdAt": "2019-07-23T02:46:46.773Z",
      "endpointArn": null,
      "eventId": "5868bfcd-54a6-4a15-9f48-76c282212659",
      "id": "826f1020-7768-4d91-b200-7ed67f682ac2",
      "mediumType": "sms",
      "updatedAt": "2020-01-28T01:04:33.206Z"
    },
    {
      "userId": "auth0|16973",
      "createdAt": "2019-07-23T11:39:29.755Z",
      "endpointArn": null,
      "eventId": "b9d52c89-5813-43be-900b-07ec3cfecdc3",
      "id": "86d34866-fd37-4a96-9344-2ea8868183c6",
      "mediumType": "email",
      "updatedAt": "2020-01-27T22:01:12.235Z"
    },
    {
      "userId": "auth0|16973",
      "createdAt": "2019-06-13T01:08:01.679Z",
      "endpointArn": null,
      "eventId": "f15ca56b-c497-4617-be28-a407087807bc",
      "id": "901af24c-384e-4d0a-8d1e-1f3bc86f3bfd",
      "mediumType": "sms",
      "updatedAt": "2020-01-28T11:10:14.600Z"
    }
  ],
  "userMobileNumber": [
    {
      "name": "Ava Anderson",
      "userId": "auth0|16973",
      "id": 18,
      "createdAt": "2019-09-29T07:40:03.496Z",
      "isActive": true,
      "phoneNumber": "376.966.8768 x95882",
      "updatedAt": "2020-01-27T20:37:32.266Z"
    },
    {
      "name": "Ava Anderson",
      "userId": "auth0|16973",
      "id": 19,
      "createdAt": "2019-10-13T02:13:45.301Z",
      "isActive": false,
      "phoneNumber": "1-379-621-8475 x10373",
      "updatedAt": "2020-01-28T01:29:54.754Z"
    },
    {
      "name": "Ava Anderson",
      "userId": "auth0|16973",
      "id": 20,
      "createdAt": "2019-07-28T05:27:01.082Z",
      "isActive": false,
      "phoneNumber": "1-101-296-1938",
      "updatedAt": "2020-01-27T23:27:41.183Z"
    },
    {
      "name": "Ava Anderson",
      "userId": "auth0|16973",
      "id": 21,
      "createdAt": "2019-03-05T10:22:12.360Z",
      "isActive": false,
      "phoneNumber": "(125) 908-2290 x71161",
      "updatedAt": "2020-01-27T21:17:57.334Z"
    },
    {
      "name": "Ava Anderson",
      "userId": "auth0|16973",
      "id": 22,
      "createdAt": "2019-02-23T20:05:31.915Z",
      "isActive": false,
      "phoneNumber": "1-704-104-1265 x28153",
      "updatedAt": "2020-01-28T13:21:58.335Z"
    }
  ]
}
```

### CODE REVIEW

File: src/events/index.js [hunk 1]
```diff
@@ -38,6 +38,31 @@ import { formatPaginatedDataFromServer } from '../utils/pagination';
...
  */
 
+/**
+ * @typedef {Object} EventUser
+ * @property {string} createdAt ISO 8601 Extended Format date/time string
+ * @property {string} email
+ * @property {string} firstName
+ * @property {string} id
+ * @property {boolean} isMachineUser
+ * @property {Array} [IOSDevices]
```
Comment (geekiam23) [inline src/events/index.js]: I do not know if this is the correct syntax to show that the api return an array of objects back.

File: src/events/index.js [hunk 2]
```diff
@@ -38,6 +38,31 @@ import { formatPaginatedDataFromServer } from '../utils/pagination';
...
+ * @property {string} [IOSDevices.createdAt] ISO 8601 Extended Format date/time string
+ * @property {boolean} [IOSDevices.isActive]
+ * @property {string} [IOSDevices.snsEndpointArn]
+ * @property {string} [IOSDevices.userId]
+ * @property {string} [IOSDevices.updatedAt] ISO 8601 Extended Format date/time string
+ * @property {string} lastName
+ * @property {Array} [userMobileNumbers]
```
Comment (geekiam23) [inline src/events/index.js]: I do not know if this is the correct syntax to show that the api return an array of objects back.

File: src/events/index.js (hunk 1 above)
Comment (geekiam23) [reply src/events/index.js]: I've had time to think about it. I think this should be {Object} but I do not know how to account for the array part

File: src/events/index.spec.js [hunk 3]
```diff
@@ -429,6 +429,70 @@ describe('Events', function() {
...
+    context('the user ID is provided', function() {
+      let userFromServerAfterFormat;
+      let userFromServerBeforeFormat;
+      let expectedUserId;
+      let promise;
+      let request;
+      let toCamelCase;
+
+      beforeEach(function() {
+        expectedUserId = fixture.build('eventUser').id;
```
Comment (greg5green) [inline src/events/index.spec.js]: This function can get cleaned up a bit:
```
beforeEach(function() {
  userFromServerAfterFormat = fixture.build('eventUser');
  userFromServerBeforeFormat = fixture.build(
    'eventUser',
    userFromServerAfterFormat,
    { fromServer: true }
  );

  request = {
    ...baseRequest,
    get: sinon.stub().resolves(userFromServerBeforeFormat)
  };
  toCamelCase = sinon
    .stub(objectUtils, 'toCamelCase')
    .returns(userFromServerAfterFormat);

  const events = new Events(baseSdk, request, expectedHost);
  events._baseUrl = expectedHost;
  promise = events.getUserInfo(userFromServerAfterFormat.id);
});
```

**Pull request merged**